odoo/addons/base/models/ir_actions_report.py

```diff
@@ -678,23 +679,20 @@ def close_streams(streams):
         if len(streams) == 1:
             result = streams[0].getvalue()
         else:
-            try:
-                result = self._merge_pdfs(streams)
-            except utils.PdfReadError:
-                raise UserError(_("One of the documents, you try to merge is encrypted"))
+            result = self._merge_pdfs(streams)
```

Comment:
> > Future tasks resulting from this PR:
> 
> Will you be creating these tasks if / when this is merged?

To be determined, I'll DM you the details.

> 
> > Odoo has a custom wrapper around PdfFileReader and PdfFileWriter in tools/pdf.py that's used in a couple of places. Check if it can be obsoleted by pikepdf.
> 
> Seems like they will need to be replaced / reimplemented for pikepdf, browsing through the docs I can't really find an API to easily manipulate embedded files. In fact, given [pikepdf/pikepdf@249bb93](https://github.com/pikepdf/pikepdf/commit/249bb938b5602d222a678db56f2cb43ba22f74ea) this might outright not be an option:
> 
> > `Pdf.attach()` has been removed, because the interface as implemented had no way to deal with existing attachments.

It's not a drop-in replacement indeed, so that would need to be investigated. Good catch on the attachments. As it happens, the traceback I mentioned when directly uploading a vendor bill through accounting happens in `OdooPdfFileReader.getAttachments()`. So if I want to fix that bug, I would need to figure out how to deal with attachments in pikepdf (for reading at least). I was planning to implement that fix next, so it might be a good test case. Maybe I'll also learn a bit more about how writing attachments would work by doing that.

Reply:
I agree, I'll check what kind of exceptions it could raise.